src/stepFunctions/commands/executeStateMachine.ts

```diff
@@ -3,135 +3,109 @@
+import { WebviewServer } from '../../webviews/server'
 
 interface CommandMessage {
     command: string
     value?: string
 }
 
-export async function executeStateMachine(params: {
-    outputChannel: vscode.OutputChannel
-    stateMachineNode: StateMachineNode
-}) {
-    const logger: Logger = getLogger()
-    const stateMachineNode = params.stateMachineNode
-    recordStepfunctionsExecuteStateMachineView()
-
-    try {
-        const view = vscode.window.createWebviewPanel('html', 'Start Execution', vscode.ViewColumn.One, {
-            enableScripts: true,
-            retainContextWhenHidden: true,
-        })
-
-        const baseTemplateFn = _.template(BaseTemplates.SIMPLE_HTML)
-        const executeTemplate = _.template(StepFunctionsTemplates.EXECUTE_TEMPLATE)
-        const loadScripts = ExtensionUtilities.getScriptsForHtml(['executeStateMachine.js'], view.webview)
-        const loadLibs = ExtensionUtilities.getLibrariesForHtml(['vue.min.js'], view.webview)
-        const loadStylesheets = ExtensionUtilities.getCssForHtml(['executeStateMachine.css'], view.webview)
+interface InitialData {
+    arn: string
+    name: string
+    region: string
+}
 
-        view.webview.html = baseTemplateFn({
-            cspSource: view.webview.cspSource,
-            content: executeTemplate({
-                StateMachineName: stateMachineNode.details.name,
-                Scripts: loadScripts,
-                Libraries: loadLibs,
-                Stylesheets: loadStylesheets,
-            }),
-        })
+interface ExecuteStateMachineMessage extends CommandMessage, InitialData {}
 
-        view.webview.onDidReceiveMessage(
-            createMessageReceivedFunc({
-                stateMachine: stateMachineNode,
-                outputChannel: params.outputChannel,
-                onPostMessage: message => view.webview.postMessage(message),
-            }),
-            undefined,
-            globals.context.subscriptions
-        )
-    } catch (err) {
-        logger.error(err as Error)
-    }
+function isExecuteStateMachineMessage(m: CommandMessage): m is ExecuteStateMachineMessage {
+    return m.command === 'executeStateMachine'
 }
 
-function createMessageReceivedFunc({
-    stateMachine: stateMachine,
-    outputChannel,
-}: {
-    stateMachine: StateMachineNode
-    outputChannel: vscode.OutputChannel
-    onPostMessage(message: any): Thenable<boolean>
-}) {
-    const logger: Logger = getLogger()
+const VueWebview = compileVueWebview({
+    id: 'remoteInvoke',
+    title: localize('AWS.executeStateMachine.title', 'Start Execution'),
+    webviewJs: 'stepFunctionsExecuteStateMachineVue.js',
+    cssFiles: ['executeStateMachine.css'],
+    commands: {
+        handler: function (message: CommandMessage | ExecuteStateMachineMessage) {
+            handleMessage(this, message)
+        },
+    },
+    start: (init: InitialData) => init,
+    retainContextWhenHidden: true, // TODO: necessary? Holdover from old impl...
```

Comment:
This webview actually has very little state if I'm using it correctly...it doesn't even print final output in the output channel. I'll give it a kick without this and see.

Reply:
Seems like it mostly just uses `data` to track whether the user is trying to read from a file or manually input JSON.
The file reading part could be broken tbh since it's doing it all within the webview. There's an input field with type "file" but I've never tested it.

Reply:
Made a few adjustments but the way this is implemented on the Vue side causes an issue: the `watch` function will always blank the JSON field if a file is loaded.